controller/internal/enforcer/applicationproxy/applicationproxy.go

```diff
@@ -382,13 +383,20 @@ func serviceTypeToApplicationListenerType(serviceType policy.ServiceType) protom
 	}
 }
 
-func buildCaches(exposedServices, dependentServices policy.ApplicationServicesList) (map[string]*urisearch.APICache, map[string]*urisearch.APICache, map[string]*x509.Certificate, [][]byte) {
+func buildCaches(exposedServices, dependentServices policy.ApplicationServicesList) (map[string]*urisearch.APICache, map[string]*urisearch.APICache, map[string]*x509.Certificate, [][]byte, map[int]string) {
 	apicache := map[string]*urisearch.APICache{}
```

Comment:
Shouldn’t you use an lru cache? Those maps may grow forever

Reply:
No. they get replaced with every policy update.